app/models/topic.rb

```diff
@@ -159,7 +159,7 @@ def set_post_forum_id
   end
 
   def count_user_posts_for_counter_cache
-    @user_posts = posts.where.not(user_id: nil).group(:user_id).count
+    @user_posts = posts.unscoped.where.not(user_id: nil).group(:user_id).count
```

Comment:
Maybe unscope only the order?

Reply:
If possible and relatively easy, I'd unscope only the order because otherwise we could be unscoping something else by accident 🤔